src/mainwindow.py

```diff
@@ -3,170 +3,315 @@
 import io
 from random import randint, choice, sample
 import asyncio
-import time
 import aiohttp
 import os
 from pygame import mixer
 
 
 class Tinder:
     def __init__(self):
+        # setup for pygame mixer
         mixer.init()
+
+        # getting the directory folder for use later when opening files
         self.dir = os.path.dirname(os.path.realpath(__file__))
+
+        # setting up the tkinter root
         self.root = tk.Tk()
         self.root.title("Cat Tinder")
         self.root.geometry("400x500")
         self.root.minsize(400, 500)
         self.root.maxsize(400, 500)
         self.root.configure(background='black')
         self.root.protocol("WM_DELETE_WINDOW", self.on_closing)
-        #self.root.bind('<Motion>', self.motion)
+
+        # getting screen width and height for use with teleporting window/jumpscare
         self.screen_x = self.root.winfo_screenwidth()
         self.screen_y = self.root.winfo_screenheight()
+
+        # setting class variables to be used later
         self.jumpscare = False
         self.loop = asyncio.get_event_loop()
         self.session = None
-        self.images = list()
-        
-        
+        self.cats = list()
+
     def start(self):
+        '''Starts the Tinder application'''
+
+        # getting a cache of cat info
         self.loop.run_until_complete(self.get_cache())
-        self.new_image()
 
+        # starting the program loop
+        self.new_image()
 
     async def get_cache(self):
-        print("refreshing image cache")
-        t = time.time()
+        '''Gets a cache of cat data and adds it to the self.cats list'''
+
+        # if we haven't created a session yet, do so
         if not self.session:
             self.session = aiohttp.ClientSession()
+
+        # Run 10 times to get 10 cats
         for i in range(10):
+            # initialize a dict of cat data
             cat_data = dict()
-            if randint(1,10) == 5 and i:
-                image_number = randint(1,10)
-                im = Image.open(os.path.join(self.dir, os.path.join("res", os.path.join("images", f"{image_number}.jpg"))))
+
+            # randomly make jumpscares happen, but not on the first image
+            if randint(1, 10) == 5 and i:
```

Comment:
I would just load a cache of 10 images and do the jumpscares at runtime. I think it gives more flexibility if we want to change the jumpscare dynamics.

Reply:
I feel like this way is better because then everything comes from the cache, meaning you only have to work with the cache in the other function instead of doing actual work.